src/Adapter/Cart/CommandHandler/AddCartRuleToCartHandler.php

```diff
@@ -114,7 +114,7 @@ public function handle(AddCartRuleToCartCommand $command)
     private function validateCartRule(CartRule $cartRule, Cart $cart): ?string
     {
         Context::getContext()->cart = $cart;
-        $previousCartRules = $cart->getCartRules();
+        $previousCartRules = $cart->getCartRules(CartRule::FILTER_ACTION_ALL, false);
```

Comment:
Hi @sowbiba 

we were wondering if this modification is still relevant? Initially, this PR fixed the issue via the command handler, then during the discussion, we went for a different approach that is handled via the BO front and JS.

But what about this modification? Is it still needed or is it a relic of the previous implementation? Or said differently should this be cleaned? or should we add some behat test to validate this modification is important and not modified later because it solves a specific issue?